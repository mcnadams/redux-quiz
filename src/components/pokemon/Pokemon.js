import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

function Pokemon({ pokemon }) {
  return (
    <Link to={`/${pokemon._id}`}>
      <img src={pokemon.url_image}/>
      <p>{pokemon.pokemon}</p>
    </Link>
  );
}

Pokemon.propTypes = {
  pokemon: PropTypes.shape({
    _id: PropTypes.string.isRequired,
    pokemon: PropTypes.string.isRequired,
    color_1: PropTypes.string.isRequired,
    color_2: PropTypes.string.isRequired,
    url_image: PropTypes.string.isRequired,
  })
};

export default Pokemon;
