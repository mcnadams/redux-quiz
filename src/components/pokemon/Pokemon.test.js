import { shallow } from 'enzyme';
import React from 'react';
import Pokemon from './Pokemon';

describe('Pokemon component', () => {
  it('renders Pokemon', () => {
    const wrapper = shallow(<Pokemon pokemon={{
      _id: '2',
      pokemon: 'bob',
      color_1: '#f0987f',
      color_2: 'NA',
      url_image: '/path',
    }}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
