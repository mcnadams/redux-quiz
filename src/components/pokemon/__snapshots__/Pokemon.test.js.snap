// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pokemon component renders Pokemon 1`] = `
<Link
  to="/2"
>
  <img
    src="/path"
  />
  <p>
    bob
  </p>
</Link>
`;
